src/InstrumentationEngine/ProfilerManager.cpp

```diff
@@ -963,28 +961,28 @@ HRESULT CProfilerManager::Initialize(
         return CORPROF_E_PROFILER_CANCEL_ACTIVATION;
     }
 
-    // Take the lock that protects the raw profiler callback and the instrumentation methods. This keeps the collection from changing out from under the iterator
+    CComPtr<ICorProfilerCallback2> pCallback;
 
-    CCriticalSectionHolder lock(&m_cs);
+    shared_ptr<CProfilerCallbackHolder> pProfilerCallbackHolder = atomic_load(&m_profilerCallbackHolder);
```

Comment:
do we want to have explicit memory_order flags for these calls? I don't know what the defaults are.

Reply:
atomic_load is equivalent to atomic_load_explicit(p, std::memory_order_seq_cst)
atomic_store is equivalent to atomic_store_explicit(p, r, memory_order_seq_cst)

Reading seq_cst ordering (https://en.cppreference.com/w/cpp/atomic/memory_order#Sequentially-consistent_ordering),
I am a bit concerned about this note:

`Total sequential ordering requires a full memory fence CPU instruction on all multi-core systems. This may become a performance bottleneck since it forces the affected memory accesses to propagate to every core.`

Reply:
I was able to get confirmation that the default flags meet performance requirements.